Build failed with `-DCAPSTONE_BUILD_DIET=ON` option

```
[12/151] Building C object CMakeFiles/capstone.dir/cs.c.o
/home/aya/Source/capstone/cs.c:550:12: warning: ‘str_replace’ defined but not used [-Wunused-function]
  550 | static int str_replace(char *result, char *target, const char *str1, char *str2)
      |            ^~~~~~~~~~~
[17/151] Building C object CMakeFiles/capstone.dir/arch/AArch64/AArch64InstPrinter.c.o
FAILED: CMakeFiles/capstone.dir/arch/AArch64/AArch64InstPrinter.c.o 
/usr/bin/cc -DCAPSTONE_ARM64_SUPPORT -DCAPSTONE_ARM_SUPPORT -DCAPSTONE_BPF_SUPPORT -DCAPSTONE_DIET -DCAPSTONE_EVM_SUPPORT -DCAPSTONE_HAS_ARM -DCAPSTONE_HAS_ARM64 -DCAPSTONE_HAS_BPF -DCAPSTONE_HAS_EVM -DCAPSTONE_HAS_M680X -DCAPSTONE_HAS_M68K -DCAPSTONE_HAS_MIPS -DCAPSTONE_HAS_MOS65XX -DCAPSTONE_HAS_POWERPC -DCAPSTONE_HAS_RISCV -DCAPSTONE_HAS_SH -DCAPSTONE_HAS_SPARC -DCAPSTONE_HAS_SYSZ -DCAPSTONE_HAS_TMS320C64X -DCAPSTONE_HAS_TRICORE -DCAPSTONE_HAS_WASM -DCAPSTONE_HAS_X86 -DCAPSTONE_HAS_XCORE -DCAPSTONE_M680X_SUPPORT -DCAPSTONE_M68K_SUPPORT -DCAPSTONE_MIPS_SUPPORT -DCAPSTONE_MOS65XX_SUPPORT -DCAPSTONE_PPC_SUPPORT -DCAPSTONE_RISCV_SUPPORT -DCAPSTONE_SH_SUPPORT -DCAPSTONE_SPARC_SUPPORT -DCAPSTONE_SYSZ_SUPPORT -DCAPSTONE_TMS320C64X_SUPPORT -DCAPSTONE_TRICORE_SUPPORT -DCAPSTONE_USE_SYS_DYN_MEM -DCAPSTONE_WASM_SUPPORT -DCAPSTONE_X86_SUPPORT -DCAPSTONE_XCORE_SUPPORT -I/home/aya/Source/capstone/include -std=gnu99 -Wunused-function -Warray-bounds -Wunused-variable -Wparentheses -Wint-in-bool-context -MD -MT CMakeFiles/capstone.dir/arch/AArch64/AArch64InstPrinter.c.o -MF CMakeFiles/capstone.dir/arch/AArch64/AArch64InstPrinter.c.o.d -o CMakeFiles/capstone.dir/arch/AArch64/AArch64InstPrinter.c.o -c /home/aya/Source/capstone/arch/AArch64/AArch64InstPrinter.c
In file included from /home/aya/Source/capstone/arch/AArch64/AArch64InstPrinter.c:2752:
/home/aya/Source/capstone/arch/AArch64/AArch64GenAsmWriter.inc: In function ‘getMnemonic’:
/home/aya/Source/capstone/arch/AArch64/AArch64GenAsmWriter.inc:13833:21: error: ‘OpInfo0’ undeclared (first use in this function); did you mean ‘OpInfo1’?
13833 |   Bits |= (uint64_t)OpInfo0[opcode] << 0;
      |                     ^~~~~~~
      |                     OpInfo1
/home/aya/Source/capstone/arch/AArch64/AArch64GenAsmWriter.inc:13833:21: note: each undeclared identifier is reported only once for each function it appears in
/home/aya/Source/capstone/arch/AArch64/AArch64GenAsmWriter.inc:13835:28: error: ‘AsmStrs’ undeclared (first use in this function)
13835 |         SStream_concat0(O, AsmStrs+(Bits & 16383)-1);
      |                            ^~~~~~~
/home/aya/Source/capstone/arch/AArch64/AArch64InstPrinter.c: At top level:
/home/aya/Source/capstone/arch/AArch64/AArch64InstPrinter.c:54:19: warning: ‘get_op_access’ defined but not used [-Wunused-function]
   54 | static cs_ac_type get_op_access(cs_struct *h, unsigned int id, unsigned int index)
      |                   ^~~~~~~~~~~~~
[21/151] Building C object CMakeFiles/capstone.dir/arch/PowerPC/PPCInstPrinter.c.o
FAILED: CMakeFiles/capstone.dir/arch/PowerPC/PPCInstPrinter.c.o 
/usr/bin/cc -DCAPSTONE_ARM64_SUPPORT -DCAPSTONE_ARM_SUPPORT -DCAPSTONE_BPF_SUPPORT -DCAPSTONE_DIET -DCAPSTONE_EVM_SUPPORT -DCAPSTONE_HAS_ARM -DCAPSTONE_HAS_ARM64 -DCAPSTONE_HAS_BPF -DCAPSTONE_HAS_EVM -DCAPSTONE_HAS_M680X -DCAPSTONE_HAS_M68K -DCAPSTONE_HAS_MIPS -DCAPSTONE_HAS_MOS65XX -DCAPSTONE_HAS_POWERPC -DCAPSTONE_HAS_RISCV -DCAPSTONE_HAS_SH -DCAPSTONE_HAS_SPARC -DCAPSTONE_HAS_SYSZ -DCAPSTONE_HAS_TMS320C64X -DCAPSTONE_HAS_TRICORE -DCAPSTONE_HAS_WASM -DCAPSTONE_HAS_X86 -DCAPSTONE_HAS_XCORE -DCAPSTONE_M680X_SUPPORT -DCAPSTONE_M68K_SUPPORT -DCAPSTONE_MIPS_SUPPORT -DCAPSTONE_MOS65XX_SUPPORT -DCAPSTONE_PPC_SUPPORT -DCAPSTONE_RISCV_SUPPORT -DCAPSTONE_SH_SUPPORT -DCAPSTONE_SPARC_SUPPORT -DCAPSTONE_SYSZ_SUPPORT -DCAPSTONE_TMS320C64X_SUPPORT -DCAPSTONE_TRICORE_SUPPORT -DCAPSTONE_USE_SYS_DYN_MEM -DCAPSTONE_WASM_SUPPORT -DCAPSTONE_X86_SUPPORT -DCAPSTONE_XCORE_SUPPORT -I/home/aya/Source/capstone/include -std=gnu99 -Wunused-function -Warray-bounds -Wunused-variable -Wparentheses -Wint-in-bool-context -MD -MT CMakeFiles/capstone.dir/arch/PowerPC/PPCInstPrinter.c.o -MF CMakeFiles/capstone.dir/arch/PowerPC/PPCInstPrinter.c.o.d -o CMakeFiles/capstone.dir/arch/PowerPC/PPCInstPrinter.c.o -c /home/aya/Source/capstone/arch/PowerPC/PPCInstPrinter.c
In file included from /home/aya/Source/capstone/arch/PowerPC/PPCInstPrinter.c:1236:
/home/aya/Source/capstone/arch/PowerPC/PPCGenAsmWriter.inc: In function ‘printInstruction’:
/home/aya/Source/capstone/arch/PowerPC/PPCGenAsmWriter.inc:6138:22: error: ‘AsmStrs’ undeclared (first use in this function)
 6138 |   SStream_concat0(O, AsmStrs+(Bits & 16383)-1);
      |                      ^~~~~~~
/home/aya/Source/capstone/arch/PowerPC/PPCGenAsmWriter.inc:6138:22: note: each undeclared identifier is reported only once for each function it appears in
In file included from /home/aya/Source/capstone/arch/PowerPC/PPCInstPrinter.c:1235:
/home/aya/Source/capstone/arch/PowerPC/PPCGenRegisterName.inc: At top level:
/home/aya/Source/capstone/arch/PowerPC/PPCGenRegisterName.inc:7:20: warning: ‘getRegisterName’ defined but not used [-Wunused-function]
    7 | static const char *getRegisterName(unsigned RegNo)
      |                    ^~~~~~~~~~~~~~~
/home/aya/Source/capstone/arch/PowerPC/PPCInstPrinter.c:1147:14: warning: ‘stripRegisterPrefix’ defined but not used [-Wunused-function]
 1147 | static char *stripRegisterPrefix(const char *RegName)
      |              ^~~~~~~~~~~~~~~~~~~
[27/151] Building C object CMakeFiles/capstone.dir/arch/X86/X86Mapping.c.o
/home/aya/Source/capstone/arch/X86/X86Mapping.c:1716:13: warning: ‘xchg_mem’ defined but not used [-Wunused-function]
 1716 | static bool xchg_mem(unsigned int opcode)
      |             ^~~~~~~~
[34/151] Building C object CMakeFiles/capstone.dir/arch/ARM/ARMDisassembler.c.o
ninja: build stopped: subcommand failed.
```

--- a/arch/AArch64/AArch64GenAsmWriter.inc
+++ b/arch/AArch64/AArch64GenAsmWriter.inc
@@ -17,6 +17,7 @@ static uint64_t getMnemonic(MCInst *MI, SStream *O, unsigned int opcode) {
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Woverlength-strings"
 #endif
+
 #ifndef CAPSTONE_DIET
   static const char AsmStrs[] = {
   /* 0 */ "sha1su0\t\0"
@@ -1131,6 +1132,8 @@ static uint64_t getMnemonic(MCInst *MI, SStream *O, unsigned int opcode) {
   /* 8739 */ "ld1w\t{\0"
   /* 8746 */ "st1w\t{\0"
 };
+#endif
+
 #ifdef __GNUC__
 #pragma GCC diagnostic pop
 #endif
@@ -7480,7 +7483,6 @@ static uint64_t getMnemonic(MCInst *MI, SStream *O, unsigned int opcode) {
     138527510U,	// anonymous_5384
     138527511U,	// anonymous_5385
   };
-#endif
 
   static const uint32_t OpInfo1[] = {
     0U,	// PHI
@@ -13832,7 +13834,9 @@ static uint64_t getMnemonic(MCInst *MI, SStream *O, unsigned int opcode) {
   uint64_t Bits = 0;
   Bits |= (uint64_t)OpInfo0[opcode] << 0;
   Bits |= (uint64_t)OpInfo1[opcode] << 32;
+#ifndef CAPSTONE_DIET
 	SStream_concat0(O, AsmStrs+(Bits & 16383)-1);
+#endif
 	return Bits;
 
 }

--- a/arch/AArch64/AArch64InstPrinter.c
+++ b/arch/AArch64/AArch64InstPrinter.c
@@ -2481,6 +2481,7 @@ static void printMatrixTileVector(MCInst *MI, unsigned OpNum, SStream *O, bool I
 	MCOperand *RegOp = MCInst_getOperand(MI, OpNum);
   	// assert(MCOperand_isReg(RegOp) && "Unexpected operand type!");
 	unsigned Reg = MCOperand_getReg(RegOp);
+#ifndef CAPSTONE_DIET
 	const char *RegName = getRegisterName(Reg, AArch64_NoRegAltName);
 
 	const size_t strLn = strlen(RegName);
@@ -2499,6 +2500,7 @@ static void printMatrixTileVector(MCInst *MI, unsigned OpNum, SStream *O, bool I
 		}
 	}
 	SStream_concat0(O, RegNameNew);
+#endif
 
 	if (MI->csh->detail) {
 #ifndef CAPSTONE_DIET
@@ -2513,7 +2515,9 @@ static void printMatrixTileVector(MCInst *MI, unsigned OpNum, SStream *O, bool I
 		MI->flat_insn->detail->arm64.operands[MI->flat_insn->detail->arm64.op_count].reg = Reg;
 		MI->flat_insn->detail->arm64.op_count++;
 	}
+#ifndef CAPSTONE_DIET
 	cs_mem_free(RegNameNew);
+#endif
 }
 
 static const unsigned MatrixZADRegisterTable[] = {

--- a/arch/PowerPC/PPCGenAsmWriter.inc
+++ b/arch/PowerPC/PPCGenAsmWriter.inc
@@ -6135,7 +6135,9 @@ static void printInstruction(MCInst *MI, SStream *O)
   uint64_t Bits = 0;
   Bits |= (uint64_t)OpInfo0[opcode] << 0;
   Bits |= (uint64_t)OpInfo1[opcode] << 32;
+#ifndef CAPSTONE_DIET
   SStream_concat0(O, AsmStrs+(Bits & 16383)-1);
+#endif
 
 
   // Fragment 0 encoded into 5 bits for 19 unique commands.

